internal/talosapi/client.go

```diff
@@ -73,106 +84,126 @@ func (c *LocalClient) osVersionInfo() (*machine.VersionInfo, error) {
 	resp, err := c.api.Version(c.ctx)
 	if err != nil || len(resp.Messages) == 0 {
 		return nil, err
-	} else {
-		return resp.Messages[0].Version, nil
 	}
+
+	return resp.Messages[0].Version, nil
 }
 
+// OSVersion returns the OS version.
 func (c *LocalClient) OSVersion() string {
 	v, err := c.osVersionInfo()
 	if err != nil {
 		c.log.WithError(err).Error("error retrieving OS version information")
+
 		return "Talos"
 	}
+
 	return fmt.Sprintf("Talos %s-%s", v.Tag, v.Sha)
 }
 
+// OSVersionShort returns the short OS version.
 func (c *LocalClient) OSVersionShort() string {
 	v, err := c.osVersionInfo()
 	if err != nil {
 		c.log.WithError(err).Error("error retrieving OS version information")
+
 		return "Talos"
 	}
+
 	return fmt.Sprintf("Talos %s", v.Tag)
 }
 
+// Hostname returns the hostname.
 func (c *LocalClient) Hostname() string {
 	resp, err := c.api.MachineClient.Hostname(c.ctx, &empty.Empty{})
 	if err != nil || len(resp.Messages) == 0 {
 		c.log.WithError(err).Error("error retrieving hostname")
+
 		return ""
-	} else {
-		return resp.Messages[0].Hostname
 	}
+
+	return resp.Messages[0].Hostname
 }
 
+// NetInterfaces returns the network interfaces.
 func (c *LocalClient) NetInterfaces() (result []tboxcmds.NetInterface) {
 	addrMap := make(map[string][]*network.AddressStatusSpec)
-	addrStatusMap, err := c.api.COSI.List(c.ctx, resource.NewMetadata(
-		network.NamespaceName,
-		network.AddressStatusType,
-		"",
-		resource.VersionUndefined,
-	))
+
+	networkAddresses, err := safe.StateListAll[*network.AddressStatus](c.ctx, c.api.COSI)
```

Comment:
this is the only relevant change, just using `safe` from COSI, so that it avoids all the nasty type casting